source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py

```diff
@@ -185,13 +186,26 @@ def write_data_to_sim(self):
         """
         # write external wrench
         if self.has_external_wrench:
-            self.root_physx_view.apply_forces_and_torques_at_position(
-                force_data=self._external_force_b.view(-1, 3),
-                torque_data=self._external_torque_b.view(-1, 3),
-                position_data=None,
-                indices=self._ALL_INDICES,
-                is_global=False,
-            )
+            if self.uses_external_wrench_positions:
+                self.root_physx_view.apply_forces_and_torques_at_position(
+                    force_data=self._external_force_b.view(-1, 3),
+                    torque_data=self._external_torque_b.view(-1, 3),
+                    position_data=self._external_wrench_positions_b.view(-1, 3),
+                    indices=self._ALL_INDICES,
+                    is_global=False,
+                )
+                # Reset the external wrench after writing to the simulation.
+                self.uses_external_wrench_positions = False
+                # Fill the positions tensor with zeros to avoid applying the wrench at the same position in the next step.
+                self._external_wrench_positions_b.fill_(0.0)
```

Comment:
Hm since the external force is applied over multiple simulation steps under the environment decimation loop, resetting here will cause issues.